Remotion/Data/DomainObjects/Persistence/Configuration/IStorageSettings.cs

```diff
@@ -0,0 +1,58 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using Remotion.Data.DomainObjects.Mapping;
+
+namespace Remotion.Data.DomainObjects.Persistence.Configuration
+{
+  /// <summary>
+  /// Defines an API for configuring the storage settings.
+  /// </summary>
+  /// <threadsafety static="true" instance="true" />
+  /// <seealso cref="StorageSettings"/>
+  /// <seealso cref="DeferredStorageSettings"/>
+  /// <threadsafety static="true" instance="true"/>
+  public interface IStorageSettings
+  {
+    /// <summary>
+    /// Gets a storage provider definition based on the given class definition.
+    /// </summary>
+    StorageProviderDefinition GetStorageProviderDefinition (ClassDefinition classDefinition);
+
+    /// <summary>
+    /// Gets a storage provider definition based on the given storage group type. If null is supplied, the default storage
+    /// provider definition is returned instead.
+    /// </summary>
+    StorageProviderDefinition GetStorageProviderDefinition (Type? storageGroupTypeOrNull);
+
+    /// <summary>
+    /// Gets a storage provider definition based on the given storage provider name.
+    /// </summary>
+    StorageProviderDefinition GetStorageProviderDefinition (string storageProviderName);
+
+    /// <summary>
+    /// Gets the default storage provider definition or null if it doesn't exist.
+    /// </summary>
+    StorageProviderDefinition? DefaultStorageProviderDefinition { get; }
+
+    /// <summary>
+    /// Gets all storage provider definitions
+    /// </summary>
+    IReadOnlyCollection<StorageProviderDefinition> StorageProviderDefinitions { get; }
```

Comment:
In mikes design those were methods. I am fine with them being properties but we should move them above the methods to keep the ordering.

Reply:
The previous implementation were also properties, meaning only breaking changes is from the old class to the new one.